src/ray/core_worker/core_worker.cc

```diff
@@ -2596,8 +2596,15 @@ void CoreWorker::HandleDeleteSpilledObjects(
 
 void CoreWorker::HandleExit(const rpc::ExitRequest &request, rpc::ExitReply *reply,
                             rpc::SendReplyCallback send_reply_callback) {
+  bool own_objects = reference_counter_->OwnObjects();
```

Comment:
This is used only by worker capping, so I just made this endpoint exit only when it doesn't own objects for simplicity (I could make more complex if / else to make it more robust for general cases).